server/render/index.js

```diff
@@ -73,20 +70,34 @@ export function render( element, key = JSON.stringify( element ) ) {
 
 export function serverRender( req, res ) {
 	const context = req.context;
+	let title, metas = [], links = [];
 
 	if ( context.lang !== config( 'i18n_default_locale_slug' ) ) {
 		context.i18nLocaleScript = '//widgets.wp.com/languages/calypso/' + context.lang + '.js';
 	}
 
-	if ( config.isEnabled( 'server-side-rendering' ) &&
-		context.store &&
-		context.layout &&
-		! context.user ) {
-		context.initialReduxState = pick( context.store.getState(), 'ui', 'themes' );
-		const key = context.renderCacheKey || JSON.stringify( context.layout );
-		Object.assign( context, render( context.layout, key ) );
+	if ( config.isEnabled( 'server-side-rendering' ) && ! context.user ) {
+		if ( context.layout ) {
+			const key = context.renderCacheKey || JSON.stringify( context.layout );
+			Object.assign( context, render( context.layout, key ) );
+		}
 	}
 
+	if ( context.store ) {
+		title = getDocumentHeadFormattedTitle( context.store.getState() );
+		metas = getDocumentHeadMeta( context.store.getState() );
+		links = getDocumentHeadLink( context.store.getState() );
+
+		let reduxSubtrees = [ 'documentHead' ];
+		if ( isSectionIsomorphic( context.store.getState() ) ) {
+			reduxSubtrees = reduxSubtrees.concat( [ 'ui', 'themes' ] );
+		}
+
+		context.initialReduxState = pick( context.store.getState(), ...reduxSubtrees );
```

Comment:
Not a big deal, but `...reduxSubtrees` doesn't need to be spread. Lodash will accept an array of keys to pick as the second argument.